src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/AbstractCrudService.java

```diff
@@ -40,7 +42,20 @@ public E findById(Integer id) {
 	}
 
 	/**
+	 * Return a proxy of the object (without hitting the database). This should
+	 * only be used if it is assumed that the object really exists and where
+	 * non-existence would be an actual error.
 	 * 
+	 * @param id
+	 * @return
+	 */
+	@PostAuthorize("hasPermission(returnObject, 'READ')")
```

Comment:
Filter is used to filter collections: https://docs.spring.io/spring-security/site/docs/current/reference/html/el-access.html#el-pre-post-annotations